src/libawkward/io/json.cpp

```diff
@@ -401,18 +463,29 @@ namespace awkward {
 
   ToJsonPrettyFile::ToJsonPrettyFile(FILE* destination,
                                      int64_t maxdecimals,
-                                     int64_t buffersize)
+                                     int64_t buffersize,
+                                     const char* nan_string,
+                                     const char* infinity_string,
+                                     const char* minus_infinity_string)
       : impl_(new ToJsonPrettyFile::Impl(destination,
                                          maxdecimals,
-                                         buffersize)) { }
+                                         buffersize))
+      , nan_string_(nan_string)
+      , infinity_string_(infinity_string)
+      , minus_infinity_string_(minus_infinity_string) { }
 
   ToJsonPrettyFile::~ToJsonPrettyFile() {
     delete impl_;
   }
 
   void
   ToJsonPrettyFile::null() {
-    impl_->null();
+    if (nan_string_ != nullptr) {
+      impl_->string(nan_string_, (int64_t)strlen(nan_string_));
+    }
+    else {
+      impl_->null();
+    }
```

Comment:
For the same reason as above, please revert this.

Reply:
done